app/controllers/vm_common.rb

```diff
@@ -1051,6 +1055,26 @@ def scan_histories
     end
   end
 
+  def parent_folder_id(vm)
```

Comment:
2 notes:

1. if the vm is archived, the id is not the `parent.id.to_cid` but rather `xx-arch`.
2. Do wish we didn't lookup the ems here since we only want the `type, id` coming out.